x/oracle/types/message_oracle.go

```diff
@@ -44,3 +63,104 @@ func (m *MsgRegisterOracle) GetSigners() []sdk.AccAddress {
 	}
 	return []sdk.AccAddress{oracleAddress}
 }
+
+var _ sdk.Msg = &MsgApproveOracleRegistration{}
+
+func NewMsgApproveOracleRegistration(approve *ApproveOracleRegistration, signature []byte) *MsgApproveOracleRegistration {
+	return &MsgApproveOracleRegistration{
+		ApproveOracleRegistration: approve,
+		Signature:                 signature,
+	}
+}
+
+func (m *MsgApproveOracleRegistration) Route() string {
+	return RouterKey
+}
+
+func (m *MsgApproveOracleRegistration) Type() string {
+	return "ApproveOracleRegistration"
+}
+
+func (m *MsgApproveOracleRegistration) ValidateBasic() error {
+	if err := m.ApproveOracleRegistration.ValidateBasic(); err != nil {
+		return err
+	}
+
+	if m.Signature == nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "signature is empty")
+	}
+
+	return nil
+}
+
+func (m *MsgApproveOracleRegistration) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(m)
+	return sdk.MustSortJSON(bz)
+}
+
+func (m *MsgApproveOracleRegistration) GetSigners() []sdk.AccAddress {
+	oracleAddress, err := sdk.AccAddressFromBech32(m.ApproveOracleRegistration.ApprovedOracleAddress)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{oracleAddress}
+}
+
+func (m *ApproveOracleRegistration) ValidateBasic() error {
+	if len(m.UniqueId) == 0 {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "uniqueId is empty")
+	}
+
+	if len(m.ApprovedOracleAddress) == 0 {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "approvedOracleAddress is empty")
+	}
+
+	if len(m.TargetOracleAddress) == 0 {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "targetOracleAddress is empty")
+	}
+
+	if m.EncryptedOraclePrivKey == nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "encryptedOraclePrivKey is empty")
+	}
+
+	return nil
+}
+
+var _ sdk.Msg = &MsgUpdateOracleInfo{}
+
+func NewMsgUpdateOracleInfo(address, endpoint string, commissionRate sdk.Dec) *MsgUpdateOracleInfo {
+	return &MsgUpdateOracleInfo{
+		OracleAddress:        address,
+		Endpoint:             endpoint,
+		OracleCommissionRate: commissionRate,
+	}
+}
+
+func (m *MsgUpdateOracleInfo) Route() string {
+	return RouterKey
+}
+
+func (m *MsgUpdateOracleInfo) Type() string {
+	return "UpdateOracleInfo"
+}
+
+func (m *MsgUpdateOracleInfo) ValidateBasic() error {
```

Comment:
Same here. The validity of oracle address should be checked.
In addition, `OracleCommissionRate` should be checked when update oracle info.